### PR TITLE
Prevent constructor to be the property name

### PR DESCRIPTION
**Pull request by jannyHou**
I am closing it due to failures, will also delete the related branch later. Sorry for the inconvenience caused.

### CODE REVIEW

File: lib/model-builder.js [hunk 1]
```diff
@@ -257,6 +257,12 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
       deprecated('Property names containing a dot are not supported. ' +
         'Model: ' + className + ', property: ' + p);
     }
+
+    // Throw error if property name is 'constructor'
+    if (p === "constructor") {
+      throw new Error('Property name should not be constructor in Model: ' +
+       className);
```
Comment (superkhau) [inline lib/model-builder.js]: missing a space here

Comment (jannyHou) [inline lib/model-builder.js]: Thank you Simon, this code is not ready for review LOL,  lots of things haven't been checked. But thanks for reminding me this, I will fix it.

Comment (superkhau) [inline lib/model-builder.js]: haha no problem, just making sure you noticed. ;)

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Pull request closed without merging**